datafusion/sqllogictest/test_files/copy.slt

```diff
@@ -307,7 +317,7 @@ select * from validate_parquet_single;
 
 # copy from table to folder of compressed json files
 query IT
-COPY source_table  to 'test_files/scratch/copy/table_json_gz' (format json, compression 'gzip');
+COPY source_table  to 'test_files/scratch/copy/table_json_gz' (format json, 'json.compression' gzip);
```

Comment:
Does this PR now support syntax like  `COPY source_table to 'test_files/scratch/copy/table_json_gz' STORED AS PARQUET`? 

I am not saying it needs to / should but I thought that was one of the goals of https://github.com/apache/arrow-datafusion/issues/9369

Reply:
Hi @alamb, the PR does not support this feature, let's handle it in a separate PR to make it easier to review. Even this one is hard to review. If we agree on the syntax, I'll create an issue for the parser change.